src/main/java/org/prebid/cache/handlers/GetCacheHandler.java

```diff
@@ -51,32 +78,108 @@ public GetCacheHandler(final ReactiveRepository<PayloadWrapper, String> reposito
+                ? processProxyRequest(request, id, cacheUrl)
+                : processRequest(request, id);
+
+        return finalizeResult(responseMono, request, timerContext);
+    }
+
+    private String resolveCacheUrl(final ServerRequest request) {
+        val cacheHostParam = request.queryParam(CACHE_HOST_KEY).orElse(null);
+        if (StringUtils.isNotBlank(cacheHostParam)) {
+            return new URIBuilder()
+                    .setHost(cacheHostParam)
+                    .setPath(apiConfig.getPath())
+                    .setScheme(config.getHostParamProtocol())
+                    .toString();
+        }
+
+        return null;
+    }
+
+    private Mono<ServerResponse> processProxyRequest(final ServerRequest request,
+                                                     final String idKeyParam,
+                                                     final String cacheUrl) {
+
+        val webClient = clientsCache.computeIfAbsent(cacheUrl, WebClient::create);
+
+        return webClient.get()
+                .uri(uriBuilder -> uriBuilder.queryParam(ID_KEY, idKeyParam).build())
+                .headers(httpHeaders -> httpHeaders.addAll(request.headers().asHttpHeaders()))
+                .exchange()
+                .transform(CircuitBreakerOperator.of(circuitBreaker))
+                .timeout(Duration.ofMillis(config.getTimeoutMs()))
+                .subscribeOn(Schedulers.parallel())
+                .handle(this::validateServerResponse)
+                .flatMap(GetCacheHandler::fromClientResponse)
+                .doOnError(error -> {
+                    metricsRecorder.getProxyFailure().mark();
+                    log.info("Failed to send request: '{}', cause: '{}'",
+                            ExceptionUtils.getMessage(error), ExceptionUtils.getMessage(error));
+
+                });
+    }
+
+    private void validateServerResponse(final ClientResponse clientResponse,
+                                        final SynchronousSink<ClientResponse> sink) {
+        if (HttpStatus.OK.equals(clientResponse.statusCode())) {
+            metricsRecorder.getProxySuccess().mark();
+            sink.next(clientResponse);
+            return;
+        }
+        sink.error(new BadRequestException(
+                String.format("Unexpected status code: '%s', in response from remote server",
+                        clientResponse.statusCode())));
+
+    }
+
+    private static Mono<ServerResponse> fromClientResponse(final ClientResponse clientResponse) {
+        return ServerResponse.status(clientResponse.statusCode())
+                .headers(headerConsumer -> clientResponse.headers().asHttpHeaders().forEach(headerConsumer::addAll))
+                .body(clientResponse.bodyToMono(String.class), String.class);
+    }
 
+    private Mono<ServerResponse> processRequest(final ServerRequest request, final String keyIdParam) {
+        val normalizedId = String.format("%s%s", config.getPrefix(), keyIdParam);
```

Comment:
Same here

Reply:
Here with String.format it is more null safe way,
else I should use "" + config.getPrefix() + keyIdParam